docs/adr/2022-04-26__alerting.md

```diff
@@ -0,0 +1,295 @@
+
+<img src="./images/alerting_branch_differences.png" width="500"/>
+
+These checks need to be manually configured for specific metrics/probes. The OpMon configuration files will be extended to something like this:
+
+```toml
+[alert]
+
+[alert.crashes] # new alert
+type = "ci"
+probes = [      # list of metrics to check
+    "startup_crashes", 
+    "main_crashes", 
+    "oom_crashes"
+]
+percentiles = [50, 90]  # percentiles that should be considered
+```
+
+
+### Thresholds
+
+In some cases the expected value of a metric is known and any large deviation from that expected value is considered a critical change. Fixed thresholds can be used to specify when a value is too large or too low. See:
+
+<img src="./images/alerting_thresholds.png" width="500"/>
+
+These checks need to be manually configured for specific metrics/probes. The OpMon configuration files will be extended to something like this:
+
+```toml
+[alert]
+
+[alert.crash_diffs] # new alert
+type = "threshold"
+precentiles = [50, 90] # percentiles that should be considered
+min = [0, 0]          # upper threshold for each percentile [optional]
+max = [10, 50]        # lower threshold for each percentile [optional]
+probes = [      # list of metrics to check
+    "startup_crashes", 
+    "main_crashes", 
+    "oom_crashes"
+]
+```
+
+An `[alert]` section can be added to configuration files as well as definition files to specify alerts.
+The minimum and maximum threshold value as well as the metrics the check should be applied to can be specified.
+A list of percentiles needs to be provided that are being checked.
+
+### Deviation from historical data
+
+Users want to get notified if a metric changed unexpectedly, for example after a new version got released. See:
+
+<img src="./images/alerting_historical_diff.png" width="500"/>
+
+It is not always possible to define a specific threshold, instead previously recorded data should be used to detect significant deviations.
+
+This check is the most complicated and computation-intensive one with potentially the highest number of false positives. There are a lot of different anomaly detection algorithms out there, but for OpMon the idea is to detect these anomalies using the exponentially weighted moving average (EWMA):
+
+```math
+EWMA_t = a * v_t + (1 - a) * EWMA_{t-1}
+```
+
+* `a`: configurable weight; larger values result in more recent recorded values having a larger impact on the moving average (= they will have a higher weight)
+* `v`: value of a specific metric
+* `EWMA_0`: the starting point, which is equal to the first value recorded for the metric
+* `t`: observation day
+
+EWMA considers all historic values but puts a lot of weight on more recent values, older values are given less weight. This should ensure that spikes recorded in the past are less likely to trigger future alerts.
+
+Control limits are determined by calculating the standard deviation:
+
+```math
+stdev(v)_t = \sqrt(EWMA(v^2)_t - EWMA(v)^2_t)
```

Comment:
Do we have a source for this?

Reply:
It based on https://assets.dynatrace.com/content/dam/en/wp/Anomaly-Detection-for-Monitoring-Ruxit.pdf page 21